zio-http/jvm/src/main/scala/zio/http/netty/WebSocketAppHandler.scala

```diff
@@ -68,9 +68,8 @@ private[zio] final class WebSocketAppHandler(
   override def exceptionCaught(ctx: ChannelHandlerContext, cause: Throwable): Unit = {
     dispatch(ctx, ChannelEvent.exceptionCaught(cause))
     onComplete match {
-      case Some(promise) =>
-        promise.fail(cause)
-      case None          =>
+      case Some(promise) => promise.unsafe.done(Exit.fail(cause))
+      case None          => ()
```

Comment:
I also noticed this bug where the promise wasn't being fulfilled with the exception as it returned an effect which wasn't then being run